Sources/PackageLoading/ManifestLoader+Validation.swift

```diff
@@ -0,0 +1,346 @@
+//
+//  File.swift
+//  
+//
+//  Created by Tom Doron on 3/4/22.
+//
+
+import Basics
+import Foundation
+import PackageModel
+import TSCBasic
+
+public enum ManifestValidator {
```

Comment:
What's the advantage of making everything static here rather than instantiating a `ManifestValidator` struct with a manifest so that the manifest doesn't have to be passed around to all the functions?  I realize there's no material difference but it seems conceptually a bit cleaner if there is every, for example, derived state that the validator might want to create and keep (e.g. lazily initialized lookup tables or whatnot).

Reply:
that's a good idea!